src/server/scripts/Northrend/VioletHold/boss_lavanthor.cpp

```diff
@@ -93,39 +91,36 @@ class boss_lavanthor : public CreatureScript
             }
         }
 
-        void MoveInLineOfSight(Unit* /*who*/) override { }
-
         void UpdateAI(uint32 diff) override
         {
             if (!UpdateVictim())
                 return;
 
-            if (uiFireboltTimer <= diff)
-            {
-                DoCastVictim(SPELL_FIREBOLT);
-                uiFireboltTimer = urand(5000, 13000);
-            } else uiFireboltTimer -= diff;
+            events.Update(diff);
 
-            if (uiFlameBreathTimer <= diff)
-            {
-                DoCastVictim(SPELL_FLAME_BREATH);
-                uiFlameBreathTimer = urand(10000, 15000);
-            } else uiFlameBreathTimer -= diff;
+            if (me->HasUnitState(UNIT_STATE_CASTING))
+                return;
 
-            if (uiLavaBurnTimer <= diff)
+            switch (uint32 eventId = events.ExecuteEvent())
```

Comment:
/home/jackpoz/trinity/sources/src/server/scripts/Northrend/VioletHold/boss_lavanthor.cpp: In member function ‘virtual void boss_lavanthor::boss_lavanthorAI::UpdateAI(uint32)’:
/home/jackpoz/trinity/sources/src/server/scripts/Northrend/VioletHold/boss_lavanthor.cpp:109:28: warning: unused variable ‘eventId’ [-Wunused-variable]
             switch (uint32 eventId = events.ExecuteEvent())
                            ^